code/modules/ai/ai_holder.dm

```diff
@@ -15,12 +15,7 @@
 	var/ai_holder_type = null // Which ai_holder datum to give to the mob when initialized. If null, nothing happens.
 
 /mob/living/Initialize()
-	if(ai_holder_type)
-		ai_holder = new ai_holder_type(src)
-		if(istype(src, /mob/living/carbon/human))
-			var/mob/living/carbon/human/H = src
-			H.hud_used = new /datum/hud(H)
-			H.create_mob_hud(H.hud_used)
+	initialize_ai_holder()
```

Comment:
One thing I'll say, I think there should be slight adjustment to logic, less so for functionality or efficiency and more for clarity, and thats purely my opinion there. I think it should check if holder is there, then call initialize. And inside initialize, check if holder is there and if NOT - then isntantly return, otherwise do the thing.